src/lib.rs

```diff
@@ -314,9 +314,15 @@ impl CspList {
                 if directive.inline_check(element, type_, policy, source) == Allowed {
                     continue;
                 }
-                let report_sample = directive.value.iter().any(|t| &t[..] == "'report-sample'");
+                let sample = if directive.value.iter().any(|t| &t[..] == "'report-sample'") {
+                    let mut trimmed_sample = source.to_owned();
```

Comment:
This could be a big allocation, given how common things like webpack are. Can we slice to min(40, len) and use to_owned in that instead?